composition/src/client_component.cpp

```diff
@@ -57,13 +57,14 @@ bool Client::on_timer()
   request->a = 2;
   request->b = 3;
 
-  while (!client_->wait_for_service(1s)) {
+  if (!client_->wait_for_service(1s)) {
     if (!rclcpp::ok()) {
       fprintf(stderr,
         "add_two_ints_client was interrupted while waiting for the service. Exiting.\n");
       return false;
     }
-    fprintf(stderr, "service not available, waiting again...\n");
+    fprintf(stderr, "service not available after waiting.\n");
+    return false;
```

Comment:
I know you didn't write this function, but why does it return a bool? It looks like the timer callback is expected to return `void`: https://github.com/ros2/rclcpp/blob/989084b3de84000a15fb4bd5a7d63929001cd769/rclcpp/include/rclcpp/timer.hpp#L145-L168

Reply:
Not sure but I'll update it to be void.

Reply:
actually, for my understanding, is it an issue if the function returns bool? Even if the return code will just be ignored if called from the timer, it might provide useful info in other contexts. What if instead we rename the function to e.g. `call_service`, which in this demo would just be called by the timer, but it could be called manually if appropriate.

Reply:
It's not a problem, but it might be an issue if this code thinks that the return type will be acted on, e.g. if it is written such that returning false should stop subsequent timer callbacks, then it would be misleading to read, since the return value is being ignored.

Reply:
ok, I've made it return void to remove any chance of readers misinterpreting that the return type is considered by the timer: d8c8b40f165381bafac21b19792074a3e58eaf08